dbms/src/Storages/AlterCommands.h

```diff
@@ -12,13 +12,19 @@ namespace DB
 
 class ASTAlterCommand;
 
-/// Operation from the ALTER query (except for manipulation with PART/PARTITION). Adding Nested columns is not expanded to add individual columns.
+/// Operation from the ALTER query (except for manipulation with PART/PARTITION).
+/// Adding Nested columns is not expanded to add individual columns.
 struct AlterCommand
 {
     enum Type
     {
         ADD_COLUMN,
         DROP_COLUMN,
+
+        /// Even though this command operates on partitions, it needs global locks to prevent table alteration.
+        /// It's vulnerable to the column modification commands.
+        FREEZE_ALL,
```

Comment:
If FREEZE_ALL is executed along with ALTER COLUMN commands, it will essentially take a write lock on the table structure. Do we really need this? I think the read lock will be sufficient (the same level of locking as e.g. the SELECT statement).

Besides, FREEZE_PARTITION should be executed under the same locks as FREEZE_ALL.